### PR TITLE
 DBZ-2202 Proposed global snapshotter interface

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-2202

Contains https://github.com/debezium/debezium/pull/2361

### CODE REVIEW

**Comment by gunnarmorling** [comment]
@jpechane, could you rebase this one? Also, what's needed in order to un-draft it?

**Comment by jpechane** [comment]
@gunnarmorling It is input for our planned discussion, so we will undraft it after.

**Review by gunnarmorling — COMMENTED** [review]
Two comments inline.

File: debezium-api/src/main/java/io/debezium/spi/snapshot/Snapshotter.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ *
+ * Implementor's must return true for either {@link #shouldSnapshot()} or {@link #shouldStream()}
+ * or true for both.
+ * 
+ * @param <T> connector configuration
+ * @param <U> state provided for snapshot execution
+ * @param <V> the data collection identifier
+ */
+@Incubating
+public interface Snapshotter<T, U, V> {
```
[the remark this answers is not in the document]
Comment (jpechane) [reply debezium-api/src/main/java/io/debezium/spi/snapshot/Snapshotter.java:33]: @gunnarmorling Well, this is the main problem. We are in the API module which knows nothing about `DataCollectionId`, `CommonConnectorConfig`. The snapshot state does not have even a common interface - we can either use Map and pass raw offsets there or we can provide connector-specific typed class like `io.debezium.connector.postgresql.spi.SnapshotState`

**Pull request closed without merging**